### PR TITLE
New model for Warp songs

### PR DESCRIPTION
**Pull request by GSKirox**
Idea taken from OOTxMM's current implementation of song notes freestandings.

This PR rotates by 180° on the Z axis all the models for the warp songs, so that they're different visually from the other songs, since only 6 colors are used for a total of 12 songs.

This brings songs more in line with the spirit of freestanding models, where you're supposed to know exactly what you'll get just by looking at it.
Since choosing different colors suitable for everyone is a big debate, this idea is a quick and easy solution.

Here's a video showing what it looks like in game : https://youtu.be/fhz18xGb7Ko
Bonus picture : 
image

### CODE REVIEW

**Comment by fenhl** [comment]
> This brings songs more in line with the spirit of freestanding models, where you're supposed to know exactly what you'll get just by looking at it.

This is neither the case currently (keys, sticks/nuts vs upgrades, BGS vs giant's knife, weird egg vs pocket egg, and of course ice traps), nor do I think this should be a goal. Some amount of ambiguity can be interesting since it makes you consider multiple options.

**Comment by GSKirox** [comment]
Well, i kinda want to work on the visibility of all the items you mentioned, except Ice traps since their whole purpose is to look the same.
But i did a simple PR first to allow me to see if the dev team considers this as a goal, or not.

**Comment by fenhl** [comment]
This specific PR also has the minor issue that the upside-down notes have the flags on the wrong side. It's just a visual bug but it's still annoying.

**Comment by cjohnson57** [comment]
> It's just a visual bug but it's still annoying.

It's not a bug, it's just that the models are rotated 180 degrees and that's what you get. The goal is to turn the notes upside down, not make them actually look like real music notes lol

**Comment by r0bd0g** [comment]
I agree that the models don't have to be different. I think the way the models aren't different for songs currently is kind of dumb though. It would make more sense if only the warp songs had colours and all the child songs were white/grey like in the pause menu.

I think my biggest problem with this PR is that the upside down songs just look dumb. The flag on the correct side would help honestly.

**Comment by TreZc0** [comment]
This has one big trigger potential cause the notes look wrong this way. 
I imagine someone else already mentioned it, but when notes are flipped upside down (which they are in notation once they pass the B on a normal G treble clef), the flag is moved inside, so that it terminates under the head of the note.

EDIT: Oh, yea, was said before, but i still disagree with making the notes turn around to make them look wrong. What's the benefit of it then? These are songs after all, rendering them with an incorrect symbol kinda doesn't click.

**Comment by GSKirox** [comment]
The benefit is to provide have a visual clue when you see a song and don't know which one it is between ZL/Bolero, Song of Time/Serenade, etc.

Right now the color is identical between pairs, and there's no way to know.
There's multiple situations where's it's actually relevent to know, sometimes you see it on the colossus arch and you can't get it yet, or you peek a skultulla token that you can't reach...

I fully agree it's a crude solution, cleaner ideas would be to have either a different model (out of my scope), or color it differently (hard to choose colors suitable for everyone).

**Comment by cjohnson57** [comment]
Jeez people are way more up in arms about the notes not looking like technically correct notes than I thought lol

**Comment by TreZc0** [comment]
Sorry, that's the musician in me speaking 🙈🙈

**Comment by GSKirox** [comment]
Ok, new idea, how about making them rotate a bit differently ?

Here's what adding a perioding rotation in Z between 0 and 45° looks like : https://youtu.be/9DeYlR6sIg0

Now of course this doesn't answer the question of if we actually want them apart, but this gets rid of the silly look of them being upside down.

**Comment by GSKirox** [comment]
Thanks to the incredible work of Maple and mracsys, there's now a new and shiny (very shiny !) inverted song model !

**Comment by GSKirox** [comment]
Video courtesy of mracsys : 

https://user-images.githubusercontent.com/65768236/206893453-ea5e80ff-c75d-47a1-9331-4038493faca5.mp4

File: Patches.py [hunk 1]
```diff
@@ -93,6 +93,17 @@ def patch_rom(spoiler:Spoiler, world:World, rom:Rom):
     # Add it to the extended object table
     add_to_extended_object_table(rom, 0x195, keyring_obj_file)
 
+    # Load Key Ring model into a file
```
Comment (NeunEinser) [inline Patches.py]:
```suggestion
    # Load Warp Song model into a file
```
Copy paste mistake ;)

**Comment by r0bd0g** [comment]
The new model looks pretty good I'd say. I have a problem with the style of the two notes being so starkly different so uh...... uhhhhhhhhh suggestion to also redo the upright note?

**Pull request merged**